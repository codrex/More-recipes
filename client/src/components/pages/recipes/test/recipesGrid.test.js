import React from 'react';
import RecipesGrid from '../recipesGrid';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';

const recipe = {
  name: 'coco rice',
  category: 'breakfast',
  views: 22,
  upVotes: 22,
  downVotes: 0,
  id: 1
};

describe('Recipes grid component ', () => {
  test('render as expected without any recipe', () => {
    const component = renderer.create(
      <BrowserRouter>
        <RecipesGrid recipes={[]} toggleFav={jest.fn()} isUserFav={jest.fn()} />
      </BrowserRouter>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('render as expected with one recipe', () => {
    const component = renderer.create(
      <BrowserRouter>
        <RecipesGrid recipes={[recipe]} toggleFav={jest.fn()} isUserFav={jest.fn()} />
      </BrowserRouter>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
