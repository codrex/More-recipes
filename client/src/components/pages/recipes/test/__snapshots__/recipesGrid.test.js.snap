// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Recipes grid component  render as expected with one recipe 1`] = `
<ul
  className="grid"
>
  <li
    className="card-wrapper"
  >
    <div
      className="recipe-card"
    >
      <div
        className="recipe-card-body"
      >
        <img
          className="recipe-card-img"
          height="250"
          src="http://res.cloudinary.com/resycom/image/upload/c_fill,h_250,w_400/v1/recipes/s9ej2m9vcvagobqadebh.jpg"
          width="400"
        />
        <div
          className="recipe-card-underlay"
        >
          <p
            className="lead text-capitalize"
          >
            breakfast
          </p>
          <button
            className="btn recipe-card-underlay-btn"
            data-dismiss="modal"
            data-target=""
            data-toggle=""
            href=""
            onClick={[Function]}
            type="button"
          >
            view details
          </button>
        </div>
      </div>
      <div
        className="recipe-card-footer"
      >
        <h4
          className="ellipsis"
        >
          coco rice
        </h4>
        <div
          className="d-flex col-12 no-padding justify-content-between"
        >
          <div
            className="d-flex col-8 no-padding"
          >
            <div
              className="icon-wrapper"
              onClick={[Function]}
              onMouseEnter={[Function]}
              role="button"
              tabIndex="0"
            >
              <i
                className="icon fa fa-thumbs-up recipe-card-icon "
                data-target=""
                data-toggle=""
              >
                <span>
                  22
                </span>
              </i>
            </div>
            <div
              className="icon-wrapper"
              onClick={[Function]}
              onMouseEnter={[Function]}
              role="button"
              tabIndex="0"
            >
              <i
                className="icon fa fa-thumbs-down recipe-card-icon "
                data-target=""
                data-toggle=""
              >
                <span>
                  0
                </span>
              </i>
            </div>
            <div
              className="icon-wrapper"
              onClick={[Function]}
              onMouseEnter={[Function]}
              role="button"
              tabIndex="0"
            >
              <i
                className="icon fa fa-eye recipe-card-icon"
                data-target=""
                data-toggle=""
              >
                <span>
                  22
                </span>
              </i>
            </div>
          </div>
          <div
            className="icon-wrapper pointer"
            onClick={[Function]}
            onMouseEnter={[Function]}
            role="button"
            tabIndex="0"
          >
            <i
              className="icon fa recipe-card-icon fav-color fav fa-heart-o"
              data-target=""
              data-toggle=""
            >
              <span />
            </i>
          </div>
        </div>
      </div>
    </div>
  </li>
</ul>
`;

exports[`Recipes grid component  render as expected without any recipe 1`] = `<div />`;
